components/scheduler/README.md

```diff
@@ -3,3 +3,215 @@
+| notification_mode    | NotificationMode    | Choose whether to display a notification on the UI, or to send by email          | false    | "show_message"                                            |
+| notification_message | string              | Set the text of the notification displayed or sent                               | false    | "Thank you for scheduling!"                               |
+| notification_subject | string              | Set the subject of the notification when using `notification_mode="show_message" | false    | "Invitation"                                              |
+| recurrence           | Recurrence          | Choose whether event should repeat? TODO                                         | false    | "none"                                                    |
+| recurrence_cadence   | RecurrenceCadence[] | Set how often the event should recur.                                            | false    | ["none"]                                                  |
+| recurrence_expiry    | RecurrenceExpiry    | Set when recurring events should stop repeating.                                 | false    | null                                                      |
+| event_options        | any[]               | TODO                                                                             | false    | []                                                        |
+
+### Events
+
+You can listen to certain events from your application by adding an event listener to the component.
+
+`onError`
+Listen to error events.
+
+```js
+document
+  .querySelector("nylas-scheduler")
+  .addEventListener("onError", (event) => {
+    let { detail } = event;
+    console.log("onError: ", detail);
+  });
+```
+
+`bookedEvents`
+Listen to bookedEvents event. Responds with an list of events.
+
+```js
+document
+  .querySelector("nylas-scheduler")
+  .addEventListener("bookedEvents", (event) => {
+    let { detail } = event;
+    console.log("bookedEvents: ", detail.events);
+  });
+```
+
+`eventOptionHovered`
+Listen to eventOptionHovered event. Responds with the actively hovered event.
+
+```js
+document
+  .querySelector("nylas-scheduler")
+  .addEventListener("eventOptionHovered", (event) => {
+    let { detail } = event;
+    console.log("eventOptionHovered: ", detail.event);
+  });
+```
+
+`eventOptionUnhovered`
+Listen to eventOptionUnhovered event.
+
+```js
+document
+  .querySelector("nylas-scheduler")
+  .addEventListener("eventOptionUnhovered", (event) => {
+    let { detail } = event;
+    console.log("eventOptionUnhovered: ", detail.event);
+  });
+```
+
+`eventOptionSelected`
+Listen to eventOptionSelected event. Responds with the actively selected event.
+
+```js
+document
+  .querySelector("nylas-scheduler")
+  .addEventListener("eventOptionSelected", (event) => {
+    let { detail } = event;
+    console.log("eventOptionSelected: ", detail.event);
+  });
+```
+
+### Types
```

Comment:
Are we providing these here to give users-who-want-to-provide-their-own-data the means to do so?